File: app/src/main/java/com/example/mykotlin/step_2/KtBase34.java
package com.example.mykotlin.step_2;

public class KtBase34 {

    interface IShowResult {
        void result(String result);
    }

    public static void main(String[] args) {

        //匿名函数 -- 匿名接口实现类
        showPersonInfo("lsit", 99, 'm', "学习C语言", new IShowResult() {
            @Override
            public void result(String result) {
                System.out.println(result);
            }
        });

        //具名函数 showResultImpl
        IShowResult mShowResult = new MShowResultImpl();
        showPersonInfo("lsit", 99, 'm', "学习C语言", mShowResult);


        String a = "JAKE";
        String b = "jake";
        System.out.println(a.equals(b));

    }

    static class MShowResultImpl implements IShowResult {

        @Override
        public void result(String result) {
            System.out.println(result);
        }
    }

    static void showPersonInfo(
            String name,
            int age,
            char sex,
            String study,
            IShowResult iShowResult
    ) {
        String str = String.format("name:%s,age:%d,sex:%c,study:%s", name, age, sex, study);
        iShowResult.result(str);
    }
}

File: app/src/main/java/com/example/mykotlin/step_2/KtBase34.kt
package com.example.mykotlin.step_2


// TODO 34.Kotlin语言的匿名函数与具名函数
fun main() {

    //匿名函数 lambda 实现体
    showPersonInfo("lisi", 99, '男', "学习") {
        print("显示结果it:$it")
    }

    //具名函数
    showPersonInfo("wangwu", 90, '女', "学习C————", ::showResultImpl)
}

fun showResultImpl(result: String) {
    println("显示结果：$result")
}

inline fun showPersonInfo(
    name: String,
    age: Int,
    sex: Char,
    study: String,
    showResult: (String) -> Unit
) {
    val str = "name:$name ,age:$age ,sex:$age ,study:$study"
    showResult(str)
}



